Upgrading Next 14.2 broke how dd-trace is compiled and tries to bring in a devDependency of dd-trace

### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/pensive-star-go8s7s

### To Reproduce

We have an `instrumentation.ts` file defined in the root with a register function as follows:

```ts
if (process.env.NEXT_RUNTIME === 'nodejs') {
  await import('./instrumentation/tracer');
}
```

Inside of `tracer.ts`

```
import tracer from 'dd-trace';

if (process.env.NODE_ENV === 'production') {
  tracer.init();
}
```

### Current vs. Expected behavior

Previously when running `next dev` there were no build issues at all. Once upgrading to 14.2.0, I now receive these error messages:

```
Module not found: Can't resolve 'graphql/utilities'
```

When I look at `dd-trace` package.json, graphql is listed as a devDependency. 

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 18.17.1
  npm: 9.6.7
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.2.0 // Latest available version is detected (14.2.0).
  eslint-config-next: 14.0.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
  dd-trace: 4.20.0
Next.js Config:
  output: standalone
```


### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

_No response_

--- a/docs/02-app/02-api-reference/05-next-config-js/serverExternalPackages.mdx
+++ b/docs/02-app/02-api-reference/05-next-config-js/serverExternalPackages.mdx
@@ -41,6 +41,7 @@ Next.js includes a [short list of popular packages](https://github.com/vercel/ne
 - `canvas`
 - `cpu-features`
 - `cypress`
+- `dd-trace`
 - `eslint`
 - `express`
 - `firebase-admin`

--- a/docs/03-pages/02-api-reference/04-next-config-js/serverExternalPackages.mdx
+++ b/docs/03-pages/02-api-reference/04-next-config-js/serverExternalPackages.mdx
@@ -41,6 +41,7 @@ Next.js includes a [short list of popular packages](https://github.com/vercel/ne
 - `canvas`
 - `cpu-features`
 - `cypress`
+- `dd-trace`
 - `eslint`
 - `express`
 - `firebase-admin`

--- a/packages/next/src/lib/server-external-packages.json
+++ b/packages/next/src/lib/server-external-packages.json
@@ -22,6 +22,7 @@
   "canvas",
   "cpu-features",
   "cypress",
+  "dd-trace",
   "eslint",
   "express",
   "firebase-admin",

